components/eventing-controller/api/v1alpha1/subscription_conversion.go

```diff
@@ -136,56 +136,79 @@ func (src *Subscription) setV2ProtocolFields(dst *v1alpha2.Subscription) {
 		}
 		// webhookAuth fields
 		if src.Spec.ProtocolSettings.WebhookAuth != nil {
-			dst.Spec.Config[v1alpha2.WebhookAuthType] = src.Spec.ProtocolSettings.WebhookAuth.Type
+			if src.Spec.ProtocolSettings.WebhookAuth.Type != "" {
+				dst.Spec.Config[v1alpha2.WebhookAuthType] = src.Spec.ProtocolSettings.WebhookAuth.Type
+			}
 			dst.Spec.Config[v1alpha2.WebhookAuthGrantType] = src.Spec.ProtocolSettings.WebhookAuth.GrantType
 			dst.Spec.Config[v1alpha2.WebhookAuthClientID] = src.Spec.ProtocolSettings.WebhookAuth.ClientID
 			dst.Spec.Config[v1alpha2.WebhookAuthClientSecret] = src.Spec.ProtocolSettings.WebhookAuth.ClientSecret
 			dst.Spec.Config[v1alpha2.WebhookAuthTokenURL] = src.Spec.ProtocolSettings.WebhookAuth.TokenURL
-			dst.Spec.Config[v1alpha2.WebhookAuthScope] = strings.Join(src.Spec.ProtocolSettings.WebhookAuth.Scope, ",")
+			if src.Spec.ProtocolSettings.WebhookAuth.Scope != nil {
+				dst.Spec.Config[v1alpha2.WebhookAuthScope] = strings.Join(src.Spec.ProtocolSettings.WebhookAuth.Scope, ",")
+			}
 		}
 	}
 }
 
+func (src *Subscription) initializeProtocolSettingsIfNil() {
+	if src.Spec.ProtocolSettings == nil {
+		src.Spec.ProtocolSettings = &ProtocolSettings{}
+	}
+}
+
+func (src *Subscription) initializeWebhookAuthIfNil() {
+	src.initializeProtocolSettingsIfNil()
+	if src.Spec.ProtocolSettings.WebhookAuth == nil {
+		src.Spec.ProtocolSettings.WebhookAuth = &WebhookAuth{}
+	}
+}
+
 // setV1ProtocolFields converts the protocol-related fields from v1alpha1 to v1alpha2 Subscription version.
 func (src *Subscription) setV1ProtocolFields(dst *v1alpha2.Subscription) {
 	if protocol, ok := dst.Spec.Config[v1alpha2.Protocol]; ok {
 		src.Spec.Protocol = protocol
 	}
 
 	if currentMode, ok := dst.Spec.Config[v1alpha2.ProtocolSettingsContentMode]; ok {
-		src.Spec.ProtocolSettings = &ProtocolSettings{}
+		src.initializeProtocolSettingsIfNil()
 		src.Spec.ProtocolSettings.ContentMode = &currentMode
 	}
 	if qos, ok := dst.Spec.Config[v1alpha2.ProtocolSettingsQos]; ok {
+		src.initializeProtocolSettingsIfNil()
```

Comment:
(nit) what if we call the `initializeProtocolSettingsIfNil` and `initializeWebhookAuthIfNil` at the start of this method, instead of doing it in every if block?

Reply:
I do not want to initialise it if its not specified. If it was nil I want to retain it as nil